maths/numerical_analysis/proper_fractions.py

```diff
@@ -0,0 +1,42 @@
+def gcd(numerator: int, denominator: int) -> int:
```

Comment:
We already have tons of implementations of `greatest_common_demonintor()` so please just use https://docs.python.org/3/library/math.html#math.gcd instead.